v2/comments.go

```diff
@@ -251,32 +268,49 @@ type Arg struct {
 	// Name is the name of a named argument. This is zero-valued for positional arguments.
 	Name string
 	// Value is the string value of an argument. It has been validated to match the Type.
-	Value string
-	// Type is the type of the Value.
-	Type ArgType
+	Value Value
+}
+
+type String string
+
+func (String) Type() Type {
+	return TypeString
+}
+
+func (s String) String() string {
+	return string(s)
+}
+
+type Int int
+
+func (Int) Type() Type {
+	return TypeInt
 }
 
-// ArgType is the type of an arg.
-type ArgType string
+func (i Int) String() string {
+	return strconv.Itoa(int(i))
```

Comment:
Why are you going thru the effort to parse string->int and then return it as a string anyway?

I'm not sure I see the value of the extra abstraction here

Reply:
It made more sense when all of the types were backed by a string (`type Bool string`) :(

Removing.